### PR TITLE
dts: bindings: uart-controller: Add detailed flow control options

### PR DESCRIPTION
**Pull request by casparfriedrich**
Add options to configure all hardware flow control mechanisms provided by the UART API.

This enables user to configure their desired flow control mechanism via device tree. Like the options for parity and stop bits, this one matches the enum provided by the UART API header `uart_config_flow_control` and can be used in a similar way.

As for the RFC: `hw-flow-control` should be marked as depricated but I'm not sure what the correct way would be to do so

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Comment by casparfriedrich** [comment]
Rebased and fixed a type. Sorry for the inconvenience @dcpleung.

**Review by dcpleung — APPROVED** [review]

**Comment by casparfriedrich** [comment]
Rebased

**Comment by casparfriedrich** [comment]
rebased

**Review by dcpleung — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Comment by casparfriedrich** [comment]
Rebased

File: dts/bindings/serial/uart-controller.yaml [hunk 1]
```diff
@@ -11,6 +11,18 @@ properties:
   current-speed:
     type: int
     description: Initial baud rate setting for UART
+  flow-control:
+    type: string
+    description: |
+      Configures the flow control to be used at boot time. Enumeration id 0 for
+      none, 1 for RTS/CTS, 2 for DTR/DSR and 3 for RS-485. Default to none if
+      not specified.
+    default: "none"
```
Comment (galak) [inline dts/bindings/serial/uart-controller.yaml:20]: There shouldn't be a default here.

**Review by galak — CHANGES_REQUESTED** [review]
Would be good to see the other side of this (ie the code change that will use this property).

Also, as this is the base controller binding, it shouldn't use `default`.

**Comment by kartben** [comment]
@casparfriedrich will you be able to address @galak 's comments?

**Comment by casparfriedrich** [comment]
Thanks for the heads up! I'm quite busy at the moment but I I'll try. Not sure though if I add something ti this pr. I'm working on A driver that would make use of this.

**Pull request closed without merging**